domain/network/state/types.go

```diff
@@ -367,6 +367,33 @@ func netInterfaceToDML(
 	return devDML, dnsSearchDMLs, dnsAddressDMLs, errors.Capture(err)
 }
 
+// decodeDeviceType returns a network interface type for an identifier congruent
+// with the database lookup.
+// The caller should have retrieved the identifier from the database directly
+// but we guard against bad input anyway.
+func decodeDeviceType(kindID int64) (corenetwork.LinkLayerDeviceType, error) {
```

Comment:
Drop this, and do the join to get it from data. You define `device` inline in `getExistingLinkLayerDevices` anyway.